utils/patientmatching/rule_based_patient_matcher.bal

```diff
@@ -0,0 +1,314 @@
+            masterPatientIndexDbUser.detail().toString(), cause = masterPatientIndexDbUser, httpStatusCode = http:STATUS_BAD_REQUEST);
+        }
+        if masterPatientIndexDbPassword is error {
+            string errorMessage = "Error occurred while getting the masterPatientIndexDbPassword from the config.json file.";
+            return r4:createFHIRError(errorMessage, r4:ERROR, r4:TRANSIENT_EXCEPTION,
+            masterPatientIndexDbPassword.detail().toString(), cause = masterPatientIndexDbPassword, httpStatusCode = http:STATUS_BAD_REQUEST);
+        }
+        MPIDbConfig mpiDbConfig = {
+            host: <string>masterPatientIndexHost,
+            port: <int>masterPatientIndexPort,
+            database: <string>masterPatientIndexDb,
+            username: <string>masterPatientIndexDbUser,
+            password: <string>masterPatientIndexDbPassword
+        };
+        return mpiDbConfig;
+
+    }
+
+    isolated function getSQLQuery(r4:Patient patient, RulesRecord rulesTable, json config) returns (error|string) {
+        json|error MPIColumnNames = config.MPIColumnNames;
+        json|error MPITableName = config.MPITableName;
+        if MPIColumnNames is error {
+            string errorMessage = "Error occurred while getting the MPIColumnNames from the config.json file.";
+            return r4:createFHIRError(errorMessage, r4:ERROR, r4:TRANSIENT_EXCEPTION,
+            MPIColumnNames.detail().toString(), cause = MPIColumnNames, httpStatusCode = http:STATUS_BAD_REQUEST);
+        }
+        if MPITableName is error {
+            string errorMessage = "Error occurred while getting the MPITableName from the config.json file.";
+            return r4:createFHIRError(errorMessage, r4:ERROR, r4:TRANSIENT_EXCEPTION,
+            MPITableName.detail().toString(), cause = MPITableName, httpStatusCode = http:STATUS_BAD_REQUEST);
+        }
+        json[] parameters = <json[]>MPIColumnNames;
+        string tableName = <string>MPITableName;
+        string query = "SELECT * FROM ";
+        query = string `${query} ${tableName} WHERE `;
+        int lastIndex = parameters.length() - 1;
+
+        foreach int i in 0 ... parameters.length() {
+            string fhirPathRule = rulesTable.fhirpathArray[i];
+            fhirpath:FhirPathResult fhirPathResult = fhirpath:getFhirPathResult(<map<json>>patient.toJson(), fhirPathRule);
+            if i == lastIndex {
+                string fhirPathRuleLastIndex = rulesTable.fhirpathArray[i];
+                map<anydata> resultMapLastIndex = fhirpath:getFhirPathResult(<map<json>>patient.toJson(), fhirPathRuleLastIndex);
+                if resultMapLastIndex.hasKey(RESULT) {
+                    json resultValue = <json>fhirPathResult.get(RESULT);
+                    if fhirPathResult.get(RESULT) is json[] {
+                        json[] resultJson = <json[]>fhirPathResult.get(RESULT);
+                        json resultFirstElement = resultJson[0];
```

Comment:
```suggestion
                        json resultFirstElement = resultValue[0];
```

Reply:
Fixed in c21e4992639515fea8603070c30ac067a7887220